internal/entities/payments/payments/core.go

```diff
@@ -1,12 +1,92 @@
 package payments
 
 import (
+	"errors"
 	"github.com/freshpay/internal/config"
+	"github.com/freshpay/internal/entities/payments/utilities"
+	"strings"
 )
 
-func AddPayments(payments *Payments) (err error) {
-	if err = config.DB.Table("payments").Create(payments).Error; err != nil {
+var InputPaymentsChannel =make(chan *Payments,1000)
+var ResultsPaymentsChannel=make(chan *Payments,1000)
+
+func AddPayments(payment *Payments) (err error) {
+	payment.Type=GetPaymentType(payment)
+	payment.Status="processing"
+	payment.ID="paymt_"+utilities.RandomString(14)
+	//err=ValidityCheck(payment)
+	//if err!=nil{
+	//	return err
+	//}
+	InputPaymentsChannel <-payment
```

Comment:
add validations:
- if user has enough balance to transfer money
- if bank accs. exist